modifiers/implicit_predicate_abstractor.h

```diff
@@ -55,9 +55,9 @@ class ImplicitPredicateAbstractor : public Abstractor
                          const smt::TermVec & new_preds,
                          smt::TermVec & out);
 
- protected:
   void do_abstraction() override;
```

Comment:
What do you think about adding an `abstracted_` flag that's checked later to ensure that the abstraction was done before using the object? The checks could be in `assert` so they're removed in release mode. Made a similar comment for axiom enumerator.

Reply:
good idea! going to change it now